src/main/webapp/app/course/competencies/prerequisite.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Prerequisite, PrerequisiteResponseDTO } from 'app/entities/prerequisite.model';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, map } from 'rxjs';
+import { CourseCompetency } from 'app/entities/competency.model';
+import { convertDateFromServer } from 'app/utils/date.utils';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class PrerequisiteService {
+    private resourceURL = 'api';
+
+    constructor(private httpClient: HttpClient) {}
+
+    getAllPrerequisitesForCourse(courseId: number): Observable<Prerequisite[]> {
+        return this.httpClient.get<PrerequisiteResponseDTO[]>(`${this.resourceURL}/courses/${courseId}/competencies/prerequisites`, { observe: 'response' }).pipe(
+            map((resp) => {
+                if (!resp.body) {
+                    return [];
+                }
+                return resp.body.map((prerequisiteDTO) => this.convertResponseDTOToPrerequisite(prerequisiteDTO));
+            }),
+        );
+        //TODO: send title to entityTitleService when we allow prerequisite detail view.
+    }
+
+    deletePrerequisite(prerequisiteId: number, courseId: number) {
+        return this.httpClient.delete<void>(`${this.resourceURL}/courses/${courseId}/competencies/prerequisites/${prerequisiteId}`, { observe: 'response' });
+    }
+
+    importPrerequisites(prerequisiteIds: number[], courseId: number): Observable<Prerequisite[]> {
+        return this.httpClient
+            .post<PrerequisiteResponseDTO[]>(`${this.resourceURL}/courses/${courseId}/competencies/prerequisites/import`, prerequisiteIds, { observe: 'response' })
+            .pipe(
+                map((resp) => {
+                    if (!resp.body) {
+                        return [];
+                    }
+                    return resp.body.map((prerequisiteDTO) => this.convertResponseDTOToPrerequisite(prerequisiteDTO));
+                }),
+            );
+    }
+
+    /**
+     * Converts a PrerequisiteResponseDTO to a Prerequisite
+     * It converts the softDueDate to dayjs and the linkedCourseCompetencyDTO to a CourseCompetency
+     * @param prerequisiteDTO PrerequisiteResponseDTO
+     * @return the Prerequisite
+     */
+    convertResponseDTOToPrerequisite(prerequisiteDTO: PrerequisiteResponseDTO): Prerequisite {
```

Comment:
this could be made static (optional: could be moved into Prerequisite by changing it to a class)